go/engine/login_offline.go

```diff
@@ -0,0 +1,181 @@
+package engine
+
+import (
+	"errors"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type LoginOffline struct {
+	libkb.Contextified
+}
+
+func NewLoginOffline(g *libkb.GlobalContext) *LoginOffline {
+	return &LoginOffline{
+		Contextified: libkb.NewContextified(g),
+	}
+}
+
+// Name is the unique engine name.
+func (e *LoginOffline) Name() string {
+	return "LoginOffline"
+}
+
+// Prereqs returns the engine prereqs.
+func (e *LoginOffline) Prereqs() Prereqs {
+	return Prereqs{}
+}
+
+// RequiredUIs returns the required UIs.
+func (e *LoginOffline) RequiredUIs() []libkb.UIKind {
+	return []libkb.UIKind{}
+}
+
+// SubConsumers returns the other UI consumers for this engine.
+func (e *LoginOffline) SubConsumers() []libkb.UIConsumer {
+	return nil
+}
+
+func (e *LoginOffline) Run(ctx *Context) error {
+	if err := e.run(ctx); err != nil {
+		return err
+	}
+
+	e.G().Log.Debug("LoginOffline success, sending login notification")
+	e.G().NotifyRouter.HandleLogin(string(e.G().Env.GetUsername()))
+	e.G().Log.Debug("LoginOffline success, calling login hooks")
+	e.G().CallLoginHooks()
+
+	return nil
+}
+
+func (e *LoginOffline) run(ctx *Context) error {
```

Comment:
every engine Run starts with a trace (in engine/engine.go:RunEngine), so I thought that would be sufficient.  Would you like something more?